### PR TITLE
add several OCSP response vectors

### PR DESCRIPTION
**Pull request by reaperhulk**
refs #6297

### CODE REVIEW

File: docs/development/test-vectors.rst [hunk 1]
```diff
@@ -569,6 +569,12 @@ X.509 OCSP Test Vectors
   ``CT Certificate SCTs`` single extension, from the SwissSign OCSP responder.
 * ``x509/ocsp/ocsp-army.deps.mil-resp.der`` - An OCSP response containing
   multiple ``SINGLERESP`` values.
+* ``x509/ocsp/resp-response-type-unknown-oid.der`` - An OCSP response with
+  an unknown OID for response type. The signature on this response is invalid.
+* ``x509/ocsp/resp-successful-no-response-bytes.der`` - An OCSP request with
+  a successful response type but the response bytes are missing.
+* ``x509/ocsp/resp-unknown-response-type.der`` - An OCSP response with an
+  unknown response type.
```
Comment (alex) [inline docs/development/test-vectors.rst:577]: This should be unknon response status, not unknown response type, right?

Comment (reaperhulk) [reply docs/development/test-vectors.rst:577]: No this covers the first item in your list of three.

Comment (alex) [reply docs/development/test-vectors.rst:577]: No it doesn't :P The first item, `resp-response-type-unknown-oid.der` does that.

Comment (reaperhulk) [reply docs/development/test-vectors.rst:577]: Oh, that’s what I get for reading this on mobile. Yeah looks like I misnamed the last item.

**Review by alex — APPROVED** [review]

**Pull request merged**